sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs

```diff
@@ -13,16 +13,22 @@ public partial class TableClient
         public virtual Azure.Data.Tables.Sas.TableSasBuilder GetSasBuilder(string rawPermissions, System.DateTimeOffset expiresOn) { throw null; }
         public virtual Azure.Response<System.Collections.ObjectModel.ReadOnlyDictionary<string, object>> Insert(System.Collections.Generic.IDictionary<string, object> entity, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.Response<System.Collections.ObjectModel.ReadOnlyDictionary<string, object>>> InsertAsync(System.Collections.Generic.IDictionary<string, object> entity, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<T>> InsertAsync<T>(T entity, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) where T : Azure.Data.Tables.TableEntity, new() { throw null; }
+        public virtual Azure.Response<T> Insert<T>(T entity, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) where T : Azure.Data.Tables.TableEntity, new() { throw null; }
```

Comment:
Do we want to enforce this inheritance structure?

Reply:
Outside of ETag, all the properties on `TableEntity` are required for any entity. Do you have any specific concerns?

Reply:
I wonder if it's an artificial limitation and we could've checked required properties at runtime and allowed any class structures customers wants.

Reply:
Could you expand any thoughts you have around what kind of class structures this structure may inhibit? To your point, we have the same required properties problem with the dictionary, so it would be more consistent to allow the same freedom. 

My thinking was that if we can provide some helpful guide-rails, we should.

Reply:
As a somewhat convoluted example: if I have a type that I can't change but want to be able to store into a table I can subclass it and add required properties. But if we require `TableEntity` to be a base class it's suddenly not possible.

Reply:
I think I'll defer that for now and raise it as a topic during review.